lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2166,11 +2217,32 @@ postAccountPublicKey
     -> ApiT DerivationIndex
     -> ApiPostAccountKeyData
     -> Handler ApiAccountKey
-postAccountPublicKey ctx (ApiT wid) (ApiT ix) (ApiPostAccountKeyData (ApiT pwd) extd) = do
+postAccountPublicKeyShelley ctx (ApiT wid) (ApiT ix) (ApiPostAccountKeyData (ApiT pwd) extd) = do
+    withWorkerCtx @_ @s @k ctx wid liftE liftE $ \wrk -> do
+        k <- liftHandler $ W.readPublicAccountKey @_ @s @k wrk wid pwd ix
+        pure $ ApiAccountKey (xPubtoBytes extd $ getRawKey k) extd
+
+xPubtoBytes :: KeyFormat -> XPub -> ByteString
+xPubtoBytes = \case
+    Extended -> xpubToBytes
+    NonExtended -> xpubPublicKey
+
+postAccountPublicKeyShared
+    :: forall ctx s k n.
+        ( s ~ SharedState n k
+        , ctx ~ ApiLayer s k
+        , HardDerivation k
+        , WalletKey k
+        )
+    => ctx
+    -> ApiT WalletId
+    -> ApiT DerivationIndex
+    -> ApiPostAccountKeyData
+    -> Handler ApiAccountKeyShared
+postAccountPublicKeyShared ctx (ApiT wid) (ApiT ix) (ApiPostAccountKeyData (ApiT pwd) extd) = do
     withWorkerCtx @_ @s @k ctx wid liftE liftE $ \wrk -> do
```

Comment:
I suggest combining them and adding a hrp prefix string as a type parameter or something like that.

Reply:
done